### PR TITLE
worktrunk: add module

### PR DESCRIPTION
**Pull request by eveeifyeve**
### Description

This pr reopens #8919, which I am happy to take over and maintain as I will be using this home-manager module everyday.

It improves the previous pr by:
- Adding more integrations with shells (zsh, fish, bash & nushell).
- Adding more tests like for shell integrations by splitting them up separately.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -A dev --run treefmt`.

- [x] Code tested through `nix run .#tests -- test-all` or
      `nix-shell --pure tests -A run.all`. (Not passing due to a firefox issue)

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [x] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [x] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

### CODE REVIEW

File: tests/modules/programs/worktrunk/bash.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    worktrunk.enable = true;
+    worktrunk.enableBashIntegration = true;
+    bash.enable = true;
+  };
+
+  nmt.script = ''
+    assertFileExists home-files/.bashrc
+    assertFileContains \
+      home-files/.bashrc \
+      'eval "$(@worktrunk@ config shell init bash)"'
```
Comment (rycee) [inline tests/modules/programs/worktrunk/bash.nix]: Seems this may work now but when I tried running the tests it wants


```suggestion
      'eval "$(@worktrunk@/bin/wt config shell init bash)"'
```

instead.

File: modules/misc/news/2026/05/2026-05-01_20-41-52.nix [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  time = "2026-05-01T10:41:52+00:00";
+  condition = true;
+  message = ''
+    A new modules is available: 'programs.worktrunk'.
```
Comment (rycee) [inline modules/misc/news/2026/05/2026-05-01_20-41-52.nix]: Please add a brief paragraph explaining what worktrunk is.

```suggestion
    A new modules is available: 'programs.worktrunk'.
    
    {brief explanation here}
```

File: tests/modules/programs/worktrunk/example-settings.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    assertFileExists home-files/.config/worktrunk/config.toml
+    assertFileContains \
+      home-files/.config/worktrunk/config.toml \
+      'skip-shell-integration-prompt = true'
+    assertFileContains \
+      home-files/.config/worktrunk/config.toml \
+      '[post-start]'
+    assertFileContains \
+      home-files/.config/worktrunk/config.toml \
+      'copy = "wt step copy-ignored"'
```
Comment (rycee) [inline tests/modules/programs/worktrunk/example-settings.nix]: It may be nicer to simply assert the content of the entire generated file?
```suggestion
    assertFileExists home-files/.config/worktrunk/config.toml
    assertFileContent \
      home-files/.config/worktrunk/config.toml \
      ${builtins.toFile "worktrunk-config.toml" ''
        {content goes here}
      ''}
```

**Comment by eveeifyeve** [comment]
I am back to doing this pr, however since worktrunk has added skills but the main problem I run into is ideally we want to wait for https://github.com/nix-community/home-manager/pull/9247 which can be another pr since it is not in-scope for this pr.

**Comment by eveeifyeve** [comment]
Hi @rycee, just following up on this one. All of your previous comments have been addressed, and the changes are ready for your review whenever you have a moment. Thank you for your time to review this so far.

File: modules/misc/news/2026/05/2026-05-01_20-41-52.nix [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  time = "2026-05-01T10:41:52+00:00";
+  condition = true;
```
Comment (Malix-Labs) [inline modules/misc/news/2026/05/2026-05-01_20-41-52.nix:3]: could potentially be enhanced with a more precise heuristic like `programs.git.enable` (or even with `|| osConfig.programs.git.enable` added, which i don't know if it's possible or desirable)

**Review by Malix-Labs — APPROVED** [review]

**Comment by Malix-Labs** [comment]
> I am back to doing this pr, however since worktrunk has added skills but the main problem I run into is ideally we want to wait for #9247 which can be another pr since it is not in-scope for this pr.

yes, but then you could also make a stacked PR on top of #9247 + #9226 if you want to prepare it

File: modules/programs/worktrunk.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  cfg = config.programs.worktrunk;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  meta.maintainers = [ lib.maintainers.eveeifyeve ];
+
+  options.programs.worktrunk = {
+    enable = lib.mkEnableOption "worktrunk";
+
+    package = lib.mkPackageOption pkgs "worktrunk" { };
```
Comment (khaneliman) [inline modules/programs/worktrunk.nix]: suggestion: allow configuration without installing Worktrunk

This module only installs the package and writes settings, so users should be able to manage configuration for an existing installation.

```suggestion
    package = lib.mkPackageOption pkgs "worktrunk" { nullable = true; };
```

Apply the package-list guard below too. Add a test with `package = null` and nonempty settings that verifies the config is generated without installing Worktrunk.

File: modules/programs/worktrunk.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        };
+      '';
+      description = ''
+        Configuration written to `$XDG_CONFIG_HOME/worktrunk/config.toml`.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ cfg.package ];
```
Comment (khaneliman) [inline modules/programs/worktrunk.nix]: suggestion: skip package installation when the package is null

This guard completes the nullable-package change while leaving configuration generation enabled.

```suggestion
    home.packages = lib.optional (cfg.package != null) cfg.package;
```

File: modules/programs/worktrunk.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      inherit (tomlFormat) type;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          skip-shell-integration-prompt = true;
+          post-start = {
+            copy = "wt step copy-ignored";
+          };
+        };
+      '';
```
Comment (khaneliman) [inline modules/programs/worktrunk.nix]: suggestion: use a plain attribute set for the example

These are all literal values, so `lib.literalExpression` is unnecessary.

```suggestion
      example = {
        skip-shell-integration-prompt = true;
        post-start = {
          copy = "wt step copy-ignored";
        };
      };
```

File: tests/modules/programs/worktrunk/default.nix [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+  worktrunk-example-settings = ./example-settings.nix;
```
Comment (khaneliman) [inline tests/modules/programs/worktrunk/default.nix:2]: issue (blocking): stub Worktrunk in the Darwin tests

Both tests enable Worktrunk on Darwin, but `tests/darwinScrublist.nix` does not include it. The Darwin test harness therefore uses the real package.

Please add this entry after `"wezterm"` in `tests/darwinScrublist.nix`

```nix
    "worktrunk"
```

File: modules/misc/news/2026/05/2026-05-01_20-41-52.nix (hunk 4 above)
Comment (eveeifyeve) [reply modules/misc/news/2026/05/2026-05-01_20-41-52.nix:3]: For a new module in HM news, I think it is always made true.
